src/pages/home/Home.js

```diff
@@ -1,55 +1,33 @@
-/* eslint-disable react/no-array-index-key */
-import React, { useEffect, useReducer } from 'react';
-import { RowSection, Banner } from '../../components';
+import React, { useEffect, useState } from 'react';
+import { RowSection, Banner, Footer } from '../../components';
 import { getProducts } from '../../services/flixycartApi';
-import { productReducer } from './reducers/productReducer';
-import { loadProductActions } from '../../constants/actions';
 import './Home.scss';
 
 const Home = () => {
-  const [productState, dispatch] = useReducer(productReducer, {
-    loading: true,
-    product: [],
-  });
+  const [products, setProducts] = useState(null);
 
   useEffect(() => {
-    document.title = 'Flixycart | Your store for latest books';
-    const { REQUEST_GET_PRODUCTS, REQUEST_GET_PRODUCTS_FAILURE, REQUEST_GET_PRODUCTS_SUCCESS } =
-      loadProductActions;
-
-    dispatch({ type: REQUEST_GET_PRODUCTS });
-    getProducts()
-      .then((res) => {
-        dispatch({
-          type: REQUEST_GET_PRODUCTS_SUCCESS,
-          payload: res.data.result,
-        });
-      })
-      .catch((err) =>
-        dispatch({
-          type: REQUEST_GET_PRODUCTS_FAILURE,
-          payload: err.message,
-        })
-      );
+    const fetchProducts = async () => {
```

Comment:
if this method is inside useffect  and you are calling just after intitalizing use IIFE Immediately invoked function

Reply:
Both using IIFE or calling functions this way is okay, and are not anti-pattern.